pkg/util/informermanager/informermanager.go

```diff
@@ -0,0 +1,287 @@
+package informermanager
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/dynamic/dynamicinformer"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	fedcorev1a1 "github.com/kubewharf/kubeadmiral/pkg/apis/core/v1alpha1"
+	fedcorev1a1informers "github.com/kubewharf/kubeadmiral/pkg/client/informers/externalversions/core/v1alpha1"
+	"github.com/kubewharf/kubeadmiral/pkg/client/listers/core/v1alpha1"
+	schemautil "github.com/kubewharf/kubeadmiral/pkg/controllers/util/schema"
+	"github.com/kubewharf/kubeadmiral/pkg/util/tools"
+)
+
+type informerManager struct {
+	lock sync.RWMutex
+
+	started bool
+
+	client      dynamic.Interface
+	ftcInformer fedcorev1a1informers.FederatedTypeConfigInformer
+
+	eventHandlerGenerators []*EventHandlerGenerator
+
+	gvrMapping *tools.BijectionMap[string, schema.GroupVersionResource]
+
+	informers                 map[string]informers.GenericInformer
+	informerStopChs           map[string]chan struct{}
+	eventHandlerRegistrations map[string]map[*EventHandlerGenerator]cache.ResourceEventHandlerRegistration
+	lastAppliedFTCsCache      map[string]map[*EventHandlerGenerator]*fedcorev1a1.FederatedTypeConfig
+
+	queue  workqueue.Interface
+	logger klog.Logger
+}
+
+func NewInformerManager(client dynamic.Interface, ftcInformer fedcorev1a1informers.FederatedTypeConfigInformer) InformerManager {
+	manager := &informerManager{
+		lock:                      sync.RWMutex{},
+		started:                   false,
+		client:                    client,
+		ftcInformer:               ftcInformer,
+		eventHandlerGenerators:    []*EventHandlerGenerator{},
+		gvrMapping:                tools.NewBijectionMap[string, schema.GroupVersionResource](),
+		informers:                 map[string]informers.GenericInformer{},
+		informerStopChs:           map[string]chan struct{}{},
+		eventHandlerRegistrations: map[string]map[*EventHandlerGenerator]cache.ResourceEventHandlerRegistration{},
+		lastAppliedFTCsCache:      map[string]map[*EventHandlerGenerator]*fedcorev1a1.FederatedTypeConfig{},
+		queue:                     workqueue.NewRateLimitingQueue(workqueue.DefaultItemBasedRateLimiter()),
```

Comment:
We don't need to construct a rate limiting queue if we're only using `workqueue.Interface`

Reply:
Oops, I meant to use the rate limiting features